### PR TITLE
Unify methods in SeatRestController.

### PR DESCRIPTION
**Pull request by MatthijsKok**
This change shouldn't break the frontend, since frontend doesn't implement admin methods.

### CODE REVIEW

File: src/main/java/ch/wisv/areafiftylan/seats/controller/SeatRestController.java [hunk 1]
```diff
@@ -51,59 +48,59 @@ public SeatRestController(SeatService seatService, UserService userService) {
...
-     */
-    @PreAuthorize("hasRole('ADMIN')")
-    @GetMapping(value = "/{group}/{number}", params = "admin")
-    Seat getSeatByGroupAndNumberAdmin(@PathVariable String group, @PathVariable int number) {
-        return seatService.getSeatBySeatGroupAndSeatNumber(group, number);
+    @GetMapping
+    MappingJacksonValue getAllSeats(@RequestParam(value = "admin", required = false) boolean admin, @AuthenticationPrincipal User user) {
+        MappingJacksonValue result = new MappingJacksonValue(seatService.getAllSeats());
+        if (!admin || !user.getAuthorities().contains(Role.ROLE_ADMIN)) {
+            System.out.println("Setting Public View...");
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/seats/controller/SeatRestController.java]: Leftover debug statement?

File: src/main/java/ch/wisv/areafiftylan/seats/controller/SeatRestController.java [hunk 2]
```diff
@@ -51,59 +48,59 @@ public SeatRestController(SeatService seatService, UserService userService) {
...
      */
-    @JsonView(View.Public.class)
     @PreAuthorize("isAuthenticated()")
     @GetMapping("/{group}")
-    SeatmapResponse getSeatGroupByName(@PathVariable String group) {
-        return seatService.getSeatGroupByName(group);
+    MappingJacksonValue getSeatGroupByName(@PathVariable String group, @RequestParam(value = "admin", required = false) boolean admin, @AuthenticationPrincipal User user) {
+        MappingJacksonValue result = new MappingJacksonValue(seatService.getSeatGroupByName(group));
+        if (!admin || !user.getAuthorities().contains(Role.ROLE_ADMIN)) {
+            System.out.println("Setting Public View...");
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/seats/controller/SeatRestController.java]: And here

File: src/main/java/ch/wisv/areafiftylan/seats/controller/SeatRestController.java [hunk 3]
```diff
@@ -51,59 +48,59 @@ public SeatRestController(SeatService seatService, UserService userService) {
...
-    @PreAuthorize("hasRole('ADMIN')")
-    @GetMapping(value = "/{group}", params = "admin")
-    SeatmapResponse getSeatGroupByNameAdminView(@PathVariable String group) {
-        return seatService.getSeatGroupByName(group);
+    @PreAuthorize("isAuthenticated()")
+    @GetMapping("/{group}/{number}")
+    MappingJacksonValue getSeatByGroupAndNumber(@PathVariable String group, @PathVariable int number, @RequestParam(value = "admin", required = false) boolean admin, @AuthenticationPrincipal User user) {
+        MappingJacksonValue result = new MappingJacksonValue(seatService.getSeatBySeatGroupAndSeatNumber(group, number));
+        if (!admin || !user.getAuthorities().contains(Role.ROLE_ADMIN)) {
+            System.out.println("Setting Public View...");
```
Comment (skamoen) [inline src/main/java/ch/wisv/areafiftylan/seats/controller/SeatRestController.java]: And here

File: src/main/java/ch/wisv/areafiftylan/seats/controller/SeatRestController.java (hunk 1 above)
Comment (MatthijsKok) [inline src/main/java/ch/wisv/areafiftylan/seats/controller/SeatRestController.java]: Whoops will fix

**Pull request merged**